### DISCUSSION TITLE
Azure File Sync - pliki usunięte po stronie cloud endpoint'a nie znikają na server endpoint'cie

### DISCUSSION THREAD

**Discussion by araneus-pl** [AzureCommunityPL/group#124] [Pomoc] [answered]
Buduję rozwiązanie integracyjne oparte na usłudze Azure File Sync. Pliki wrzucone na lokalny udział sieciowy są synchronizowane do Azure File Shares za pomocą Azure File Sync'a, w chmurze są następnie przetwarzane za pomocą Logic Apps and Azure Functions.

Tutaj trafiłem na problem: plik usunięty po stronie cloud endpoint'a nie jest usuwany po stronie server endpoint'a, nawet przy ręcznym wymuszeniu synchronizacji za pomocą Invoke-AzStorageSyncChangeDetection . Wszystkie inne zmiany propagują się poprawnie. 

FileSyncErrorsReport.ps1 nie wyrzuca żadnych błędów. Spotkał się ktoś z czymś takim?

**Comment by miksci** [comment]
nie blokuje Ci coś ruchu na on-prem? np. firewall?

**Reply by araneus-pl** [reply]
Nie, nic mi nie blokuje.

**Comment by araneus-pl** [comment, marked as answer]
Dostałem odpowiedź na Q&A Microsoft'u:

Operacje usunięcie pliku po stronie cloud enpointa wykrywane są raz na 24h, a Invoke-AzStorageSyncChangeDetection działa tylko na zmiany plików, a nie na ich usunięcie.

Szczegóły tutaj:

https://docs.microsoft.com/en-us/answers/questions/141057/azure-file-sync-files-not-deleted-on-server-endpoi.html

**Comment by miksci** [comment]
zastanawiam się nad innym scenariusze:
- user usuwa plik na on-prem - powiedzmy przypadkowo
- masz file sync i z clouda plik też znika
- ale masz też backup na storage account i robisz odzysk - powiedzmy, że z opcją odzyskania do pierwotnej lokalizacji
- w cloud plik odzyskany się pojawi
- a co z on-prem, tam się zapewnie nie pojawi :)
